api/envoy/api/v2/auth/cert.proto

```diff
@@ -336,6 +336,12 @@ message UpstreamTlsContext {
   //
   // Defaults to 1, setting this to 0 disables session resumption.
   google.protobuf.UInt32Value max_session_keys = 4;
+
+  // High risky setting! If true, client will downgrade to plain text connection.
+  // Should only be used for opportunistic encryption.
+  // .. attention::
+  // TLS renegotiation is considered insecure and shouldn't be used unless absolutely necessary.
+  bool allow_downgrade = 5;
```

Comment:
I'm not a fan of putting this option here, and make the connection creating logic to interfere from here to create Permissive transport socket. It makes it impossible to compile out permissive transport socket. Rather than this, can you take a look on tap transport socket to see how it wraps other transport socket with explicit configuration? That way it is more flexible that the permissive transport socket can wrap other transport socket (theoretically ALTS, or the OpenSSL one lives out of tree).

Reply:
Make sense. I will make the change.